### PR TITLE
CAD-1285 calculate resource usage

### PR DESCRIPTION
**Pull request by CodiePP**
outputs CPU usage [%] and memory usage [MB] in a CSV file for all nodes.

@deepfire : in scripts/lib.sh I had to distinguish between "build" and "run" modes for `stack`. Only one of them knows about the "package:type:component" selector.

### CODE REVIEW

File: bm-timeline/src/Resources.hs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      let memusage = sortOn fst $ reverse $ map (\(t0,d) -> (TL.toStrict t0, d)) $ processMemory (length nodes) memusage0
+      let cpuusage = sortOn fst $ reverse $ map (\(t0,d) -> (TL.toStrict t0, d)) $ processCPUticks (length nodes) cputicks0
+      -- TLIO.putStrLn . TL.pack . show $ take 20 cpuusage
+      -- TLIO.putStrLn . TL.pack . show $ take 20 memusage
+      let header nm = "timestamp" : (map (\nid -> nm <> " " <> T.pack (show nid)) nodes)
+      output_csv h $ named_columns [ (header "cpu", timestamp_with_list(cpuusage))
+                                   , (header "mem", timestamp_with_list(memusage)) ]
+  return ()
+
+processMemory :: Int -> [MemResident.MemResident] -> [(Text, [Double])]
```
Comment (deepfire) [inline bm-timeline/src/Resources.hs:69]: Maybe we could introduce `newtype` wrappers for those `Double`'s to highlight their intent?

**Review by deepfire — APPROVED** [review]
Looks good!

File: bm-timeline/src/Resources.hs (hunk 1 above)
Comment (CodiePP) [reply bm-timeline/src/Resources.hs:69]: yes, right. and in the future we may need wrap it in a Maybe so we can create "empty" cells.

**Pull request merged**